### PR TITLE
CDAP-19729 : Enable upgrade for Datastream apps using state store.

### PR DESCRIPTION
**Pull request by greeshmaswaminathan**
[CDAP-19729] Enable upgrade for Datastream apps using state store. 

[CDAP-19729]: https://cdap.atlassian.net/browse/CDAP-19729?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: cdap-app-templates/cdap-etl/cdap-data-streams-base/src/main/java/io/cdap/cdap/datastreams/DataStreamsApp.java [hunk 1]
```diff
@@ -52,4 +54,25 @@ public void configure() {
...
   }
+
+  @Override
+  public boolean isUpdateSupported() {
+    return true;
+  }
+
+  @Override
+  public ApplicationUpdateResult<DataStreamsConfig>
+  updateConfig(ApplicationUpdateContext applicationUpdateContext) throws Exception {
```
Comment (albertshau) [inline cdap-app-templates/cdap-etl/cdap-data-streams-base/src/main/java/io/cdap/cdap/datastreams/DataStreamsApp.java]: nit: think it's a little more readable if the method name is on the previous line, with just the context on this line

File: cdap-app-templates/cdap-etl/cdap-data-streams-base/src/main/java/io/cdap/cdap/datastreams/DataStreamsApp.java [hunk 2]
```diff
@@ -52,4 +54,25 @@ public void configure() {
...
+    DataStreamsPipelineSpec spec = DataStreamsPipelineSpecGenerator.getDataStreamsPipelineSpec(
+      applicationUpdateContext.getApplicationSpecification());
+    if (!spec.getStateSpec().getMode().equals(DataStreamsStateSpec.Mode.STATE_STORE)) {
+      throw new UnsupportedOperationException(
+        String.format("The app %s do not support upgrade with state handling mode %s .",
```
Comment (albertshau) [inline cdap-app-templates/cdap-etl/cdap-data-streams-base/src/main/java/io/cdap/cdap/datastreams/DataStreamsApp.java]: The app %s do not -> App %s does not

**Review by albertshau — APPROVED** [review]
minor comments, otherwise lgtm

File: cdap-app-templates/cdap-etl/cdap-data-streams-base/src/main/java/io/cdap/cdap/datastreams/DataStreamsApp.java (hunk 1 above)
Comment (greeshmaswaminathan) [reply cdap-app-templates/cdap-etl/cdap-data-streams-base/src/main/java/io/cdap/cdap/datastreams/DataStreamsApp.java]: Done.

File: cdap-app-templates/cdap-etl/cdap-data-streams-base/src/main/java/io/cdap/cdap/datastreams/DataStreamsApp.java (hunk 2 above)
Comment (greeshmaswaminathan) [reply cdap-app-templates/cdap-etl/cdap-data-streams-base/src/main/java/io/cdap/cdap/datastreams/DataStreamsApp.java]: Done.

**Pull request merged**